[504] Migrate production to AKS

smneal opened the pull request:
### Context

Change required to migrate production from GOV.UK PaaS to the Teacher Services Cloud AKS platform.

This builds an image where the production AKS environment uses the proper production domains and includes the DNS changes to point `www` (for find and publish) and `api` (for publish) to the new AKS environment.

### Changes proposed in this pull request

- Removal of production input and jobs from various GitHub action workflows and actions.
- Update the production_aks environment configuration to use the `www` and `api` domains. Previously `www3` and `api3`.
- Remove the existing CNAME records that are used by the GOV.UK PaaS environment
- Add the CNAME record for `www`, `api` to point to the previously provisioned front door endpoint. This may fail first time unless the existing CNAME deletion happens quickly enough. Just re-run it.
- Update the database backup and restore workflow which backs up prod, sanitises it and restores it to QA and staging to use the AKS production database and also postgres 14.7 so the pg_dump workers properly.
- Delete the scheduled PaaS to AKS backup and restore workflow.

### Guidance to review

- Tested database backup and sanitise job of workflow [here](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5680596414/job/15395031815).

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

RMcVelia commented on .github/actions/deploy_v2/action.yml:
```diff
@@ -49,7 +49,7 @@ runs:
-            echo "deploy_url=https://api3.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
+            echo "deploy_url=https://api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
           else
             echo "deploy_url=https://${app_environment}2.api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
```
${app_environment}2.api to ${app_environment}.api ?

smneal replied:
Good shout

RMcVelia commented:
FYI, for register I cleaned up config/settings and config/environments straight after all envs migrated, to get rid of duplicate files.
i.e. https://github.com/DFE-Digital/register-trainee-teachers/pull/3445
or keep _aks and delete the other ones.

smneal commented on terraform/custom_domains/environment_domains/workspace_variables/find_production.tfvars.json:
```diff
@@ -4,6 +4,7 @@
       "resource_group_name": "s189p01-fttdomains-rg",
       "domains": [
+        "www",
         "www3"
```
I have been removing these after the migration as tidy-up to reduce the time waiting for the resources to be destroyed during the actual migration window.

smneal commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -437,7 +414,7 @@ jobs:
+    needs: [deploy-aks-production]
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
```
Updated this based on the comments

smneal commented:
> FYI, for register I cleaned up config/settings and config/environments straight after all envs migrated, to get rid of duplicate files. i.e. [DFE-Digital/register-trainee-teachers#3445](https://github.com/DFE-Digital/register-trainee-teachers/pull/3445) or keep _aks and delete the other ones.

Thanks Rick. I've added this to the post-migration activity on the migration plan.
